src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -26,21 +26,18 @@ static void report_progress_cb(event::Channel *channel, const Uptane::Target &ta
 }
 
 SotaUptaneClient::SotaUptaneClient(Config &config_in, const std::shared_ptr<INvStorage> &storage_in,
-                                   std::shared_ptr<HttpInterface> http_in, std::shared_ptr<ReportQueue> report_queue_in,
+                                   std::shared_ptr<HttpInterface> http_in,
                                    std::shared_ptr<event::Channel> events_channel_in)
     : config(config_in),
       uptane_manifest(config, storage_in),
       storage(storage_in),
       http(std::move(http_in)),
-      report_queue(std::move(report_queue_in)),
+      uptane_fetcher(new Uptane::Fetcher(config, http)),
+      report_queue(new ReportQueue(config, http)),
```

Comment:
This looks promising. However, this means it is no longer possible to *not* have a report queue. That's probably fine, but in the report queue itself, we have this sort of annoying thing where we only know not to use it if we get 404s when sending reports, and we don't store that information. We dump a trace-level log message every time we get a 404. Maybe there's a better way to do this. At any rate, the default case is clearer here, and should work just fine, so that is good.

Reply:
We were always creating one, so I see no reason to keep it optional. For the issue you describe, we probably can have a connectivity flag somewhere in sotauptaneclient.